src/js/components/stock-movement-wizard/outbound/PickPage.jsx

```diff
@@ -641,39 +705,80 @@ class PickPage extends Component {
     }
   }
 
+  onSaveAndExit(values) {
+    if (!this.validateEmptyPicks(values)) {
+      this.showEmptyPicksErrorMessage(values);
+      return;
+    }
+
+    window.location = STOCK_MOVEMENT_URL.show(values.stockMovementId);
+  }
+
   render() {
-    const { itemFilter } = this.state;
+    const {
+      showOnlyErroredItems,
+      itemFilter,
+      showAlert,
+      alertMessage,
+    } = this.state;
     const { showOnly } = this.props;
+
     return (
       <Form
         onSubmit={values => this.nextPage(values)}
         mutators={{ ...arrayMutators }}
         initialValues={this.state.values}
         render={({ handleSubmit, values }) => (
           <div className="d-flex flex-column">
-            <AlertMessage show={this.state.showAlert} message={this.state.alertMessage} danger />
+            <AlertMessage show={showAlert} message={alertMessage} danger />
             { !showOnly ?
               <span className="buttons-container">
                 <FilterInput
                   itemFilter={itemFilter}
                   onChange={e => this.setState({ itemFilter: e.target.value })}
                   onClear={() => this.setState({ itemFilter: '' })}
                 />
+                <button
+                  type="button"
+                  onClick={() => {
+                    this.validateEmptyPicks(values);
```

Comment:
Is this required here?

Reply:
Yup, because this function sets `hasError` - an indicator of whether we have to hide/show a row